lib/utils/AuthContext.js

```diff
@@ -145,7 +177,7 @@ class AuthContext extends Context {
 	 * @param {string|undefined} tokenId
 	 * @param {string|undefined} tokenPassword
 	 */
-	setContext(newContext, tokenId, tokenPassword) {
+	setContext(newContext, tokenId, tokenPassword, token) {
```

Comment:
I think time to refactor this function for acceptin only one `settings object` argument :)